device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1170,7 +1198,14 @@ private void acknowledgeReceivedMessage(IotHubTransportMessage receivedMessage)
                         {
                             Object context = correlationCallbackContexts.get(correlationId);
                             callback.onResponseAcknowledged(receivedMessage, context);
+
+                            correlationCallbackContexts.remove(correlationId);
```

Comment:
Could this be moved down to the same block that clears the other correlation callback states on 1207 and 1208?

Reply:
I prefer to leave it here since it's possible that, for a particular correlationId, we only have key-value object in the `correlationCallbacks` map but no object in the `correlationCallbackContexts` map if there's no CorrelatingMessageCallbackContext for that message ([ref](https://github.com/Azure/azure-iot-sdk-java/blob/main/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java#L1816)).